### DISCUSSION TITLE
Trouble accessing Environment Variables

### DISCUSSION THREAD

**Discussion by hhirshland** [vercel/community#3645] [Help] [answered]
### Summary

I am trying to store API keys as an environment variable, but am having trouble getting this to work.

Currently, I have a .env (that is included in my .gitignore) has the variable OPEN_AI_KEY=key... When running locally, my code succeeds in pulling in the variable (via: process.env.OPEN_AI_KEY).  However, when I then deploy the same code using vercel, the environment variable doesn't seem to be loading.  Note, I have added the OPEN_AI_KEY environment variable in my vercel project settings, and it is identical to the key used in my .env file.

I'm trying to understand what I am doing wrong, any help is much appreciated!

### Steps to Reproduce

Add environment variable in vercel project settings.

Deploy code, attempting to access environment variable via process.env.OPEN_AI_KEY.

**Comment by amyegan** [comment, marked as answer]
Hey @hhirshland. I wasn't able to replicate this problem in my test app. If you share your public repo or a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), that should let me see what is going wrong.

**Reply by hhirshland** [reply]
Hey @amyegan !

Here is my public repo: https://github.com/hhirshland/what-to-cook

Note, I am also noticing that I am getting a 504 error when trying to fetch from /api/getRecipes (in ingredientsModule.js) on my deployed project (but it works as expected when running locally).

Thanks for the help :)

**Reply by amyegan** [reply]
Thanks for sharing that. I changed the `/api/getRecipes` code to only log the environment variable and return a generic message (skipping the Open AI call). I also created a new project with my own fake `OPEN_AI_KEY` variable. When I ran a test deployment of this code, the key was logged from `/api/getRecipes` as expected.

I see you are also trying to log it in a component and it prints out `undefined`. That makes sense because this variable is sensitive and should not be exposed to the browser. For any values that you did want to expose to the browser, you would need to add the prefix `NEXT_PUBLIC_`. You can learn more about that in the Next docs at https://nextjs.org/docs/app/building-your-application/configuring/environment-variables#bundling-environment-variables-for-the-browser
